Identity.DataAccess/UserSeed.cs

```diff
@@ -0,0 +1,79 @@
+﻿using Identity.Core.Models;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.EntityFrameworkCore;
+
+namespace Identity.DataAccess
+{
+    public class UserSeed(ModelBuilder modelBuilder)
+    {
+        private readonly ModelBuilder _modelBuilder = modelBuilder;
+
+        public void Seed()
+        {
+            var adminRoleId = Guid.NewGuid().ToString();
+            var userRoleId = Guid.NewGuid().ToString();
+
+            //Roles
+            _modelBuilder.Entity<IdentityRole>().HasData(
+                new IdentityRole
+                {
+                    Id = adminRoleId,
+                    ConcurrencyStamp = adminRoleId,
+                    Name = "Admin",
+                    NormalizedName = "ADMIN"
+                },
+                new IdentityRole
+                {
+                    Id = userRoleId,
+                    ConcurrencyStamp = userRoleId,
+                    Name = "User",
+                    NormalizedName = "USER"
+                });
+
+            var adminId = Guid.NewGuid().ToString();
+            var adminUser = new User
+            {
+                Id = adminId,
+                Email = "admin@admin.com",
+                EmailConfirmed = true,
+                UserName = "AdminTest",
+                NormalizedEmail = "ADMIN@ADMIN.COM",
+                NormalizedUserName = "ADMINTEST"
+            };
+
+            var userId = Guid.NewGuid().ToString();
+            var regularUser = new User
+            {
+                Id = userId,
+                Email = "user@example.com",
+                EmailConfirmed = false,
+                UserName = "JoeDoe",
+                NormalizedUserName = "JOEDOE",
+                NormalizedEmail = "USER@EXAMPLE.COM"
+            };
+
+            //Add Hash Password
+            var passwordHasher = new PasswordHasher<User>();
+            adminUser.PasswordHash = passwordHasher.HashPassword(adminUser, "0451");
+            regularUser.PasswordHash = passwordHasher.HashPassword(regularUser, "F_p@raD0x?");
+
+
+            _modelBuilder.Entity<User>().HasData(
+                adminUser,
+                regularUser);
+
+            //Link Test Users with Roles
+            _modelBuilder.Entity<IdentityUserRole<string>>().HasData(
+                new IdentityUserRole<string>
+                {
+                    RoleId = adminRoleId,
+                    UserId = adminId
+                },
+                new IdentityUserRole<string>
+                {
+                    RoleId = userRoleId,
+                    UserId = userId
+                });
+        }
+    }
```

Comment:
I think it's better to put the code in separate private methods

Reply:
```suggestion
                public void Seed()
        {
            var adminRoleId = Guid.NewGuid().ToString();
            var userRoleId = Guid.NewGuid().ToString();

            //Roles
            AddRoles(adminRoleId, userRoleId);


            var adminId = Guid.NewGuid().ToString();
            var userId = Guid.NewGuid().ToString();

            AddTestData(adminId, userId);

            //Link Test Users with Roles
            GrantRole(adminRoleId, adminId);
            GrantRole(userRoleId, userId);
        }

        private void AddRoles(string adminRoleId, string userRoleId)
        {
            _modelBuilder.Entity<IdentityRole>().HasData(
                new IdentityRole
                {
                    Id = adminRoleId,
                    ConcurrencyStamp = adminRoleId,
                    Name = "Admin",
                    NormalizedName = "ADMIN"
                },
                new IdentityRole
                {
                    Id = userRoleId,
                    ConcurrencyStamp = userRoleId,
                    Name = "User",
                    NormalizedName = "USER"
                });
        }

        private void AddTestData(string adminId, string userId)
        {
            var passwordHasher = new PasswordHasher<User>();

            var adminUser = new User
            {
                Id = adminId,
                Email = "admin@admin.com",
                EmailConfirmed = true,
                UserName = "AdminTest",
                NormalizedEmail = "ADMIN@ADMIN.COM",
                NormalizedUserName = "ADMINTEST",
            };
            adminUser.PasswordHash = passwordHasher.HashPassword(adminUser, "0451");

            var regularUser = new User
            {
                Id = userId,
                Email = "user@example.com",
                EmailConfirmed = false,
                UserName = "JoeDoe",
                NormalizedUserName = "JOEDOE",
                NormalizedEmail = "USER@EXAMPLE.COM"
            };
            //Add Hash Password
            regularUser.PasswordHash = passwordHasher.HashPassword(regularUser, "F_p@raD0x?");


            _modelBuilder.Entity<User>().HasData(
                adminUser,
                regularUser);
        }

        private void GrantRole(string roleId, string userId)
        {
            _modelBuilder.Entity<IdentityUserRole<string>>().HasData(
                new IdentityUserRole<string>
                {
                    RoleId = roleId,
                    UserId = userId
                });
        }
```

Like this?

Reply:
yes